go/obsclient/obsclient.go

```diff
@@ -0,0 +1,43 @@
+package obsclient
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/obscuronet/go-obscuro/go/rpcclientlib"
+)
+
+// ObsClient requires an RPC Client and provides access to general Obscuro functionality that doesn't require viewing keys.
+//
+// The methods in this client are analogous to the methods in geth's EthClient and should behave the same unless noted otherwise.
+type ObsClient struct {
+	RPCClient rpcclientlib.Client
+}
+
+func Dial(rawurl string) (*ObsClient, error) {
+	rc, err := rpcclientlib.NewNetworkClient(rawurl)
+	if err != nil {
+		return nil, err
+	}
+	return NewObsClient(rc), nil
+}
+
+func NewObsClient(c rpcclientlib.Client) *ObsClient {
```

Comment:
Same comment here - wouldn't it be better to hide the existence of `rpcclientlib.Client` from the user, and just require an address?

Reply:
I think that's the purpose of the `Dial()`-style convenience constructors, so we'd be merging the two constructors, but I think this one is a dependency injection style constructor with no side-effects (the connection stuff has already been dealt with), nicer to work with than initialising the struct and allows a place to add any initialisation without breaking usages.

But I do agree on the Auth change you suggested, if it takes a Client and a ViewingKey in the `NewAuthObsClient` then it's nice depending on the same interface rather than the `EncRPCClient` directly, and makes it clearer how it differs from `ObsClient`.